### PR TITLE
Increase readability of the Admin Logs

### LINKED ISSUE #717: Increase the specificity of Admin Logs for events.
Currently, the admin log for when an event is created/deleted only says the name of the event and its start date. We want to change that so it says, at least, what program the event belongs to and if it is a recurring event.

### PR DESCRIPTION
**Pull request by AndersonStettner**
Did:
Changed the wording of the Admin Logs when an event is created or deleted. 
We now indicate what program the event is being created for, link to the newly created event, indicate if it is a non-program event, indicate if it is a recurring event, and, if it is a recurring event, also show the start date of the last event in the series. Changed how we redirect to the newly created event after event creation and how we redirect 

----------------------------------------------------
Files changed: 
- app/controllers/admin/routes.py
- app/logic/events.py
- app/templates/admin/adminLogs.html
----------------------------------------------------
To test: 
Create any program event, go to the Admin Logs, and make sure the text is accurate and the link takes you to the correct page.

Create any non-program event, go to the Admin Logs, and make sure the text is accurate and the link takes you to the correct page.

Create any program or non-program recurring event, go to the Admin Logs, and make sure the text is accurate and the link takes you to the correct page. Also, make sure the start date of the last event in the series is correct. 

----------------------------------------------------
Before:
image

After:
image

### CODE REVIEW

**Comment by AndersonStettner** [comment]
[Increase the specificity of Admin Logs for events. #717](https://trello.com/c/OjpxLAX5/517-increase-the-specificity-of-admin-logs-for-events-717)

File: app/controllers/admin/routes.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def createEvent(templateid, programid=None):
     if request.method == "POST":
         try:
             saveSuccess, validationErrorMessage = attemptSaveEvent(eventData, attachmentFiles)
-            createLog(f"Created event: {eventData['name']}, which had a start date of {datetime.strftime(eventData['startDate'], '%m/%d/%Y')}")
+            createLog(f"Created \"{eventData['name']}\" for {program.programName}, with a start date of {datetime.strftime(eventData['startDate'], '%m/%d/%Y')}")
```
Comment (Karina-Agliullova) [inline app/controllers/admin/routes.py]: Add colon (:) after Created to be consistent with the wording

File: app/logic/events.py [hunk 2]
```diff
@@ -54,7 +56,7 @@ def deleteEvent(eventId):
 
         event.delete_instance(recursive = True, delete_nullable = True)
 
-        createLog(f"Deleted event: {event.name}, which had a start date of {datetime.datetime.strftime(event.startDate, '%m/%d/%Y')}")
+        createLog(f"Deleted \"{event.name}\" for {program.programName}, which had a start date of {datetime.datetime.strftime(event.startDate, '%m/%d/%Y')}")
```
Comment (Karina-Agliullova) [inline app/logic/events.py]: Add a colon (:) after Deleted to be consistent with the wording. When you delete a non-program event, you have a colon, while here you don't have it

File: app/controllers/admin/routes.py [hunk 3]
```diff
@@ -95,22 +95,26 @@ def createEvent(templateid, programid=None):
...
-            if eventData['isRecurring']: 
-                lastEventInSeries = Event.get(Event.id == eventId)
-                eventId = Event.get(Event.recurringId == lastEventInSeries.recurringId)
-            return redirect(url_for("admin.eventDisplay", eventId = eventId))
+
+            if not program:
+                createLog(f"Created a non-program event: <a href=\"{url_for('admin.eventDisplay', eventId = savedEvents[0].id)}\">{savedEvents[0].name}</a>, with a start date of {datetime.strftime(eventData['startDate'], '%m/%d/%Y')}.")
+            else:
+                if len(savedEvents) > 1:
+                    createLog(f"Created a recurring event, <a href=\"{url_for('admin.eventDisplay', eventId = savedEvents[0].id)}\">{savedEvents[0].name}</a>, for {program.programName}, with a start date of {datetime.strftime(eventData['startDate'], '%m/%d/%Y')}. The last event in the series will be on {datetime.strftime(savedEvents[-1].startDate, '%m/%d/%Y')}.")
```
Comment (Karina-Agliullova) [inline app/controllers/admin/routes.py:111]: Add colons when creating and deleting events to be consistent. This is a good example of where you use a colon "Created a non-program event:"

File: app/logic/events.py [hunk 4]
```diff
@@ -52,11 +53,25 @@ def deleteEvent(eventId):
                     newEventName = recurringEvent.name
                     eventDeleted = True
 
-        event.delete_instance(recursive = True, delete_nullable = True)
+        program = event.singleProgram
+
+        if not program:
```
Comment (Karina-Agliullova) [inline app/logic/events.py]: Replace the negative (not). Switch the content of the condition

File: app/logic/events.py [hunk 5]
```diff
@@ -52,11 +53,25 @@ def deleteEvent(eventId):
...
+            createLog(f"Deleted a non-program event: \"{event.name}\", which had a start date of {datetime.datetime.strftime(event.startDate, '%m/%d/%Y')}.")
+        else:
+            createLog(f"Deleted \"{event.name}\" for {program.programName}, which had a start date of {datetime.datetime.strftime(event.startDate, '%m/%d/%Y')}.")
 
-        createLog(f"Deleted event: {event.name}, which had a start date of {datetime.datetime.strftime(event.startDate, '%m/%d/%Y')}")
+        event.delete_instance(recursive = True, delete_nullable = True)
 
 def attemptSaveEvent(eventData, attachmentFiles = None):
+    """
```
Comment (Karina-Agliullova) [inline app/logic/events.py:66]: Good docstring!

**Review by tylerpar99 — APPROVED** [review]
Looks good.

**Pull request merged**